MarsRover/Controller.cs

```diff
@@ -91,36 +116,26 @@ private Location InitialiseLocation()
 
             if (Map.HasObstacle(location)) 
             {
-                _output.WriteLine(Messages.InvalidInput);
+                _output.WriteLine(Messages.InvalidLocation);
                 InitialiseLocation();
             }
             return location;
         }
-        // user asked for string of commands
-        // user enters string of commands
-        // commands are validated
-        // commands are parsed 
-        // rover follows each command
-        // Move logic:
-        // 1/ GetTargetLocation to determine where Rover should Move to
-        // 2a/ use Map.GetSquareAtLocation in order to find Square -  ?? HasObstacle - Controller
-        // 2b/ use Map.HasObstacle on the square to check if there is an obstacle - Controller
-        // 3/ if no obstacle, set Rover.Location to the location of that square - ?? Move
-        // if there is an obstacle, Rover.Location remains the same
-        // Rover reports to user re obstacle
-        public void Run()
+
+        private List<Command> GetCommands()
         {
+            var commands = new List<Command>();
             _output.WriteLine(Messages.RoverCommands);
             var commandString = _input.ReadLine();
             var areAllCommandsValid = Validator.AreCommandsValid(commandString);
-            while(!areAllCommandsValid)
+            while (!areAllCommandsValid)
             {
                 _output.WriteLine(Messages.InvalidInput);
                 _output.WriteLine(Messages.RoverCommands);
                 commandString = _input.ReadLine();
                 areAllCommandsValid = Validator.AreCommandsValid(commandString);
             }
-            var commands = InputParser.ParseCommands(commandString);
+            return commands = InputParser.ParseCommands(commandString);

```

Comment:
Presumably `InputParser.ParseCommands(commandString)` returns a `List<Command>` type? If so, probably don't need to instantiate `commands` on line 127. Also don't need to assign on line `138`